src/compiler/checker.ts

```diff
@@ -11225,6 +11234,56 @@ namespace ts {
             }
         }
 
+        /**
+         * Determines if all declarations in the given list are function declarations, and that
+         * all the declarations have the same name. This is useful for how we report errors.
+         *
+         * For instance, if a user has
+         *
+         *      export default function foo() { }
+         *      export default function bar() { }
+         *
+         * We want to tell the user that there are multiple default exports. However, if a user has
+         *
+         *      export default function foo() { }
+         *      export default function foo() { }
+         *
+         * It is probably more accurate to say that there are duplicate function implementations.
+         * 
+         * @param declarations

```

Comment:
I don't know about our JSDoc style, but in the past I have skipped parameters that had no associated documentation. Is that allowed in JSDoc? Alternatively, instead of deleting this line, it would be nice to have some text here.